src/rebar_prv_shell.erl

```diff
@@ -331,8 +331,8 @@ reread_config(State) ->
     case find_config(State) of
         no_config ->
             ok;
-        ConfigList ->
-            _ = rebar_utils:reread_config(ConfigList),
+        Config ->
+            _ = rebar_utils:reread_config([Config]),
```

Comment:
Right. Is the problem that a flatmap was used in `rebar_file_utils`, instead of using a more nested sequence of maps in the callsite?

Say 

    apply_configs       (takes a list of config sets) 
    --> apply_config    (applies a single list of config vars)
       --> apply_var    (applies individual variables) 

Then the change could be applied without changing other callsites? It seems that the distinction was lost between one config and many configs (which would totally preserve order still) by using a flatmap to merge all individual lists into a single one.